### PR TITLE
Prefer admin defined DNS

### PR DESCRIPTION
**Pull request by matt-FFFFFF**
Small change to prefer admin defined DNS servers, that is those defined using ```hassio dns options --servers dns://1.2.3.4``` 

See #1316 for discussion.

### CODE REVIEW

**Comment by matt-FFFFFF** [comment]
This is especially important for users that block outbound DNS or users that run split DNS, using the same namespace internally as externally.

E.g. LetsEncryot cert for hass.mydomain.com, internally this resolves to the RFC 1918 address of the hassio instance, externally resolves to the public IP.

**Comment by pvizeli** [comment]
The issue is, that this property before 193 is set default to the predefined servers. Next version don't set this server.

That means, all instance use the backup dns server instead the dns server from DHCP/local system. If we solve this issue, I'm fine to change that to prefere manual DNS over local DHCP/local settings from host OS.

**Comment by matt-FFFFFF** [comment]
As 193 is out now can this be included in 194?

**Comment by pvizeli** [comment]
Nope, because only new installs are not affected from this.

However, that means only this PR needs smarter to avoid the issue, that doesn't mean we can not do it.

**Comment by matt-FFFFFF** [comment]
I have added a test to compare ```self.servers``` to ```DNS_SERVERS```. If identical then self.servers is ignored.

**Comment by pvizeli** [comment]
We should add on line `119` or so this check and clear the server list with a note that we remove that code on version 200. Good work, thanks!

**Comment by matt-FFFFFF** [comment]
Hi - I have added comments to the reset method, is that what you meant? I could not see anything on ```119```.

**Comment by pvizeli** [comment]
More like

```
async def load(...):
    ....
   # (After line 115) Fix dns server handling before 194 / Cleanup with version 200
   if DNS_SERVERS == self.servers:
      self.servers.clear()
      self.save_data()
````

**Comment by matt-FFFFFF** [comment]
Ok, done. Thanks

File: hassio/dns.py [hunk 1]
```diff
@@ -215,15 +222,28 @@ def _write_corefile(self) -> None:
             _LOGGER.error("Can't read coredns template file: %s", err)
             raise CoreDNSError() from None
 
-        # Prepare DNS serverlist: Prio 1 Local, Prio 2 Manual, Prio 3 Fallback
+        # Prepare DNS serverlist: Prio 1 Manual, Prio 2 Local, Prio 3 Fallback
+        # If Manual servers are identical to Fallback (was default before 194), then ignore manual
         local_dns: List[str] = self.sys_host.network.dns_servers or ["dns://127.0.0.11"]
+        if DNS_SERVERS == self.servers:
```
Comment (pvizeli) [inline hassio/dns.py]: That now not happens anymore with your change above. So you can fallback to your first solution

Comment (matt-FFFFFF) [reply hassio/dns.py]: Got it, thanks - removed condition in 1a8c1d5

**Pull request merged**